packages/valory/skills/abstract_round_abci/base.py

```diff
@@ -355,6 +361,55 @@ def __eq__(self, other: Any) -> bool:
         return self.payload == other.payload and self.signature == other.signature
 
 
+@dataclass(frozen=True)
+class NewBaseTxPayload(ABC, metaclass=_MetaPayload):
+    """This class represents a base class for transaction payload classes."""
+
+    sender: str
+    round_count: int = field(default=ROUND_COUNT_DEFAULT, init=False)
+
+
+@dataclass
+class NewTransaction:
+    """Class to represent a transaction for the ephemeral chain of a period."""
+
+    payload: NewBaseTxPayload
+    signature: Optional[str] = None
+
+    def __bytes__(self) -> bytes:
+        """Serialize the transaction to bytes"""
+
+        payload = {self.payload.__class__.__name__: asdict(self.payload)}
+        tx_data = dict(payload=payload, signature=self.signature)
+        encoded_data = json.dumps(tx_data, sort_keys=True).encode()
+        if sys.getsizeof(encoded_data) > MAX_READ_IN_BYTES:
+            msg = f"Transaction must be smaller than {MAX_READ_IN_BYTES} bytes"
+            raise ValueError(msg)
+        return encoded_data
+
+    @classmethod
+    def from_bytes(cls, obj: bytes) -> "NewTransaction":
+        """Deserialize the transaction from bytes"""
+
+        data = json.loads(obj.decode())
+        payload, signature = data["payload"], data["signature"]
+        cls_name, kwargs = payload.popitem()
+        round_count = kwargs.pop("round_count")
+        payload_cls = _MetaPayload.transaction_type_to_payload_cls[cls_name]
+        payload = payload_cls(**kwargs)
+        object.__setattr__(payload, "round_count", round_count)
+        return cls(payload=cast(NewBaseTxPayload, payload), signature=signature)
```

Comment:
`round_count` must be set using `object` since its own setter will raise a `dataclasses.FrozenInstanceError`.

Reply:
This should be a comment in the code.

Reply:
Why set `init=False` here:

https://github.com/valory-xyz/open-autonomy/blob/4d623d97896c7ad3675cb82de07d3dcd7099c38a/packages/valory/skills/abstract_round_abci/base.py#L369

instead of setting it to `True` and not using `object.__setattr__` but keep the `round_count` in the `kwargs`?

Reply:
please see my previous comment here: https://github.com/valory-xyz/open-autonomy/pull/1722#discussion_r1065169704 where I explain that we can otherwise not use non-default arguments in any of the concrete payload implementations.